### DISCUSSION TITLE
Lazy @cmd2.with_argparser() ?!?

### DISCUSSION THREAD

**Discussion by jedie** [python-cmd2/cmd2#1104] [General]
I used in some places a argparser that does use choices. This choices list is created dynamically and that takes a little time.
Unfortunately the list is evaluated directly at the start :(

Is there a possibility that the list is evaluated only when it is really needed?

**Comment by jedie** [comment]
Ah! Using **choices_function** or **choices_method** is the solution...

(choices_function / choices_method will be replaced with choices_provider in cmd 2.0)

**Reply by anselor** [reply]
Looks like you got it.
